lib/token.cpp

```diff
@@ -312,10 +357,10 @@ void Token::replace(Token *replaceThis, Token *start, Token *end)
     start->previous(replaceThis->previous());
     end->next(replaceThis->next());
 
-    if (end->tokensBack && *(end->tokensBack) == end) {
+    if (end->tokensFrontBack->back && end->tokensFrontBack->back == end) {
```

Comment:
this condition should be simplified. as far as I see this should work also:

    if (end->tokensFrontBack->back == end) {

Reply:
I think the if check was originally supposed to guard against null pointers.  I've fixed it so it does that properly now.